drivers/include/wiegand.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2019 Hackerspace San Salvador
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_wiegand Wiegand
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ *
+ * @brief       GPIO based Wiegand driver
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ * @{
+ *
+ * @file
+ * @brief       Driver for the Wiegand Protocol
+ *
+ * @author      Mario Gomez <mario.gomez@teubi.co>
+ */
+
+#ifndef WIEGAND_H
+#define WIEGAND_H
+
+#include <stdint.h>
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Parameters needed for device initialization
+ */
+typedef struct {
+    gpio_t d0;              /**< Wiegand D0 data line */
+    gpio_t d1;              /**< Wiegand D1 data line */
+    gpio_flank_t flank;     /**< flank detection */
+} wiegand_params_t;
+
+/**
+ * @brief   Device descriptor for a wiegand device
+ */
+typedef struct {
+    uint32_t code;                          /**< Wiegand code */
+    int16_t type;                           /**< Wiegand type */
+    bool ready;                             /**< Ready to convert */
+    int16_t last_bit_count;                 /**< Latest bit-count */
+    volatile uint32_t wg_card_temp_high;    /**< Card MSB */
+    volatile uint32_t wg_card_temp;         /**< Card LSV  */
+    volatile uint32_t wg_last_wiegand;      /**< Last card lecture */
+    volatile int16_t wg_bit_count;          /**< Number of bits read */
```

Comment:
volatile won't help you with race conditions - there is no reason for this to be `volatile`.

Reply:
Added them as volatile to prevent the compiler to optimize them at compile time. This variables are accessed by the "bit" counters at interrupt time. However looking at the code I agree that I don't think the compiler will try to do any optimizations, just trying to be extra-cautious.